daft/expressions/expressions.py

```diff
@@ -1660,7 +1660,12 @@ def name(self) -> builtins.str:
         return self._expr.name()
 
     def over(self, window: Window) -> Expression:
-        return Expression._from_pyexpr(self._expr.over(window._spec))
+        expr = self._expr.over(window._spec)
+        return Expression._from_pyexpr(expr)
+
+    def rank(self) -> Expression:
+        expr = self._expr.rank()
+        return Expression._from_pyexpr(expr)
```

Comment:
I don't think rank belongs in expressions. It also should not take in self. Let's move it to the functions module instead